app/herdbook.py

```diff
@@ -505,6 +509,314 @@ def testbreed():
     return {"offspringCOI": formatted_offspring_coi}
 
 
+@APP.route("/api/certificates/update/<i_number>", methods=["PATCH"])
+@login_required
+def update_certificate(i_number):
+    """
+    Returns an updated pdf of the individual given by `i_number`.
+    """
+    user_id = session.get("user_id", None)
+    ind = da.get_individual(i_number, user_id)
+
+    if ind is None:
+        return jsonify({"response": "Individual not found"}), 404
+
+    certificate_exists = ind.get("certificate", None)
+    form = request.json
+    uploaded = False
+
+    try:
+        present = check_certificate_s3(ind_number=ind["number"])
+        if certificate_exists and present:
+            data = get_certificate_data(ind, user_id)
+            data.update(**form)
+            pdf_bytes = get_certificate(data)
+            signed_data = sign_data(pdf_bytes)
+            uploaded = upload_certificate(
+                pdf_bytes=signed_data.getvalue(), ind_number=ind["number"]
+            )
+            ind.update(**form)
+            da.update_individual(ind, session.get("user_id", None))
+    except Exception as ex:  # pylint: disable=broad-except
+        APP.logger.info("Unexpected error while updating certificate " + str(ex))
```

Comment:
I think the "can't update paper certificate" case deserves enough treatment that the frontend can give a sensible error.

Reply:
The HTTP status code(404) returned by the exception handler is mostly for when calling the API with an individual that does not have a digital certificate(hopefully a paper one but this is not guaranteed) uploaded **or the backend is broken**. I think this will mostly happen when the user tries to update the certificate of **the wrong individual,** though. I am fine with changing this but note it will not always convey the correct information.

Reply:
Ok, after some thinking I believe we really should improve the frontend anyway and this won't be such a big thing.